crates/metrics/src/buckets.rs

```diff
@@ -0,0 +1,141 @@
+use std::{
+    collections::HashMap,
+    sync::OnceLock,
+};
+#[cfg(test)]
+use strum_macros::EnumIter;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(test, derive(EnumIter))]
+pub(crate) enum Buckets {
+    Timing,
+    GasUsed,
+    TransactionsCount,
+    Fee,
+    SizeUsed,
+}
+static BUCKETS: OnceLock<HashMap<Buckets, Vec<f64>>> = OnceLock::new();
+pub(crate) fn buckets(b: Buckets) -> impl Iterator<Item = f64> {
+    BUCKETS.get_or_init(initialize_buckets)[&b].iter().copied()
+}
+
+#[rustfmt::skip]
+fn initialize_buckets() -> HashMap<Buckets, Vec<f64>> {
+    [
+        (
+            Buckets::Timing,
+            vec![
+                0.005,
+                0.010,
+                0.025,
+                0.050,
+                0.100,
+                0.250,
+                0.500,
+                1.000,
+                2.500,
+                5.000,
+               10.000,
+            ],
+        ),
+        (
+            // Gas limit is 30_000_000.
+            Buckets::GasUsed,
+            vec![
+                    10_000.0,
+                    25_000.0,
+                    50_000.0,
+                   100_000.0,
+                   250_000.0,
+                   500_000.0,
+                 1_000_000.0,
+                 3_000_000.0,
+                 7_500_000.0,
+                15_000_000.0,
+                25_000_000.0,
+            ],
+        ),
+        (
+            // Transaction count is fixed at 65_535.
+            Buckets::TransactionsCount,
+            vec![
+                     5.0,
+                    10.0,
+                    25.0,
+                    50.0,
+                   100.0,
+                 1_000.0,
+                 5_000.0,
+                10_000.0,
+                20_000.0,
+                40_000.0,
+                60_000.0,
+            ],
+        ),
+        (
+            // 50 ETH is expected to be the realistic maximum fee for the time being.
```

Comment:
Probably should still specify that this is in Gwei. 

Also, I don't know if we should bother having so much granularity up high. When I said 50 ETH in our convo, I was thinking in terms of the Maximum conceiveable fee, well beyond what will happen any time soon. 

We should have many, much lower.

For example a single tx from testnet [here](https://app-testnet.fuel.network/tx/0x7e69f9dd4d6068cb503a5d0a7279732ae0cc22db91a153bbc80b2d2a241d9337/advanced):
```
gasCosts:{
    fee:"10",
    gasUsed:"87570"
},
```

This is during low volume, etc, but it's an idea of what we _could_ be working with, much more likely than even 0.05 ETH.

So perhaps we break it down like: 0, 100, 200, 400, 800, 1_600, etc... It's less aggressive than log_10, and gives us helpful granularity.

The highest for now can just be `52_428_800.0`? Are we okay with 20 buckets?

Reply:
> Probably should still specify that this is in Gwei.

The metric itself is explicit about the unit:
```rust
        registry.register(
            "executor_fee_per_block_gwei",
            "The total fee (gwei) paid by transactions in a block",
            fee_per_block.clone(),
        );
```

I see that we don't use gwei across the fuel-core codebase (except for the `fuel-gas-price-algorithm` crate), which I think makes "gwei" unit a default for developers.

Reply:
> So perhaps we break it down like: 0, 100, 200, 400, 800, 1_600, etc... It's less aggressive than log_10, and gives us helpful granularity.
> 
> The highest for now can just be `52_428_800.0`? Are we okay with 20 buckets?

Yes, that's a good observation, thanks. I think we're good with ~20 buckets, we can still re-adjust them once we start observing real-world data.

Updated in https://github.com/FuelLabs/fuel-core/pull/2310/commits/b41331b3b96c2e7f47b5dd6b46c33959482ce8b7